### PR TITLE
rimage: stop ignoring errors in parse_uuid()

### PR DESCRIPTION
**Pull request by marc-hb**
3 commits.

Also fix compilation with clang 15

I checked that the MTL checksums are identical before/after these code changes.

### CODE REVIEW

**Comment by marc-hb** [comment]
https://github.com/thesofproject/sof/actions/runs/6591943429/job/17911586211?pr=8366 has only DOS line endings warnings.

**Review by RanderWang — APPROVED** [review]

File: tools/rimage/src/include/rimage/toml_utils.h [hunk 1]
```diff
@@ -111,7 +111,11 @@ void parse_printable_key(const toml_table_t *table, struct parse_ctx *ctx, const
 void parse_str_key(const toml_table_t *table, struct parse_ctx *ctx, const char *key,
 		   char *dst, int capacity, int *error);
 
-void parse_uuid(char *buf, uint8_t *uuid);
+/**
+ * Parse UUID hex string into a byte array. The endianness of the output
+ * is architecture-dependent: do not use in any portable code.
+ */
+void parse_uuid(const char *buf, uint8_t *uuid);
```
Comment (pjdobrowolski) [inline tools/rimage/src/include/rimage/toml_utils.h:118]: Since Rimage is part of sof repository can we add ifdef condition for Intel adsp?

Comment (marc-hb) [reply tools/rimage/src/include/rimage/toml_utils.h:118]: Feel free to submit a follow-up PR after this one. This is one is just a minor clean-up.

File: tools/rimage/src/toml_utils.c [hunk 2]
```diff
@@ -294,8 +295,12 @@ void parse_uuid(char *buf, uint8_t *uuid)
 	struct uuid_t id;
 	uint32_t d[10];
 
+	const int parsed_uuid_fields =
 	sscanf(buf, "%08x-%04x-%04x-%02x%02x-%02x%02x%02x%02x%02x%02x", &id.d0, &d[0],
 	       &d[1], &d[2], &d[3], &d[4], &d[5], &d[6], &d[7], &d[8], &d[9]);
```
Comment (lyakh) [inline tools/rimage/src/toml_utils.c]: would be good to indent this, e.g. as
```
const int parsed_uuid_fields = sscanf(buf,
				      "%08x-%04x-%04x-%02x%02x-%02x%02x%02x%02x%02x%02x",...
```

**Review by lyakh — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**